### ISSUE TITLE
Training with the low-level/generic torch optimizer

### ISSUE THREAD

**Issue by pricexu** [UKPLab/sentence-transformers#2860]
Hi there,

This library is very convenient and easy to use. However, my question is, how can we train the sentence transformers in a more low-level way. For example, if it is possible to do:

emb = sentence_transformer(text)
x = another_module(emb)
loss = loss_func(x, label)
loss.backward()
optim.step()

My motivation is that in my setting, the training of the sentence_transformer is according to the feedback from another module.

Edited: I think this might work:

model.train()
encoding = model.tokenizer(text, padding=True, truncation=True, return_tensors='pt')
x = model(encoding)['sentence_embedding']
loss = loss_func(x, label)
loss.backward()
optim.step()

Many thanks!
-Jay
